src/d3a/models/market/two_sided_pay_as_bid.py

```diff
@@ -120,70 +171,45 @@ def accept_bid(self, bid: Bid, energy: float = None,
             f"trade rate: {trade_rate} market {market_bid.price / market_bid.energy}"
 
         orig_price = bid.original_bid_price if bid.original_bid_price is not None else bid.price
-        residual = None
+        residual_bid = None
 
         if energy <= 0:
             raise InvalidTrade("Energy cannot be negative or zero.")
         elif energy > market_bid.energy:
             raise InvalidTrade("Traded energy cannot be more than the bid energy.")
         elif energy < market_bid.energy:
-            # Partial bidding
-            energy_portion = energy / market_bid.energy
-            residual_price = (1 - energy_portion) * market_bid.price
-            residual_energy = market_bid.energy - energy
-
-            # Manually creating the bid in order to not double-charge the fee of the
-            # residual bid
-            changed_bid = Bid(
-                str(uuid.uuid4()), residual_price, residual_energy,
-                buyer, seller,
-                original_bid_price=(1 - energy_portion) * orig_price,
-                buyer_origin=market_bid.buyer_origin
-            )
-
-            self.bids[changed_bid.id] = changed_bid
-            self.bid_history.append(changed_bid)
-
-            self._notify_listeners(MarketEvent.BID_CHANGED,
-                                   existing_bid=bid, new_bid=changed_bid)
-            residual = changed_bid
-
-            revenue, fees, final_trade_rate = GridFees.calculate_trade_price_and_fees(
-                trade_offer_info, self.transfer_fee_ratio
-            )
-            self.market_fee += fees
-            final_price = energy * final_trade_rate
-            bid = Bid(bid.id, final_price, energy, buyer, seller,
-                      original_bid_price=energy_portion * orig_price,
-                      buyer_origin=bid.buyer_origin)
+            # partial bid trade
+            accepted_bid, residual_bid = self.split_bid(market_bid, energy, orig_price)
+            bid = accepted_bid
+
+            # Delete the accepted bid from self.bids:
+            try:
+                self.bids.pop(accepted_bid.id)
+            except KeyError:
+                raise BidNotFound
```

Comment:
You need to instantiate the exception class here, I am not sure that it works just with the class name (never tried it, but I have doubts that it works that way). Also an informative message does not hurt.

Reply:
Will do, thanks!